code-review.js

```diff
@@ -1,60 +1,119 @@
+"use strict";
 /**
  * A priority queue stores a list of items but each can have a numeric priority value.
  * Items with a higher priority are dequeued before items with a lower priority.
  * Implemented as a hash of arrays where the hash keys are priority values.
+ *
+ * @class
  */
-function PriorityQueue(size) {
-	this.store = {};	// keys are priorities, values are arrays of elements
+function PriorityQueue() {
+	this.queue = {};	// keys are priorities, values are arrays of elements
 	this.count = 0;
 
-	// adds an item
-	// priority must be an integer (higher value has higher priority)
+
+	/**
+	 * Adds an item
+	 * priority must be a positive integer (higher value has higher priority)
+	 * 
+	 * @param {*} value - The Item we are adding to the queue
+	 * @param {Number} priority - Positive integer
+	 */
 	this.add = function(value, priority) {
-		if (this.store[priority] == undefined)
-			this.store[priority] = [];
+		if (this.queue[priority] == undefined) {
+			this.queue[priority] = [];
+		}
 
-		this.store[priority].push(value);
+		this.queue[priority].push(value);
 		this.count++;
 	};
 
-	// returns the oldest-added value with the highest priority
-	this.Pop = function() {
-		maxKey = Math.max(Object.keys(this.store));
+	/**
+	 * Returns the oldest-added value with the highest priority
+	 * @returns {*} - value removed
+	 */
+	this.pop = function() {
+		if (this.count === 0) {
+			return false;
+		}
+		var maxKey = Math.max.apply(null, this.getAllPriorities());
+		var poppedValue = this.queue[maxKey].shift();
 		this.count--;
-		return this.store[maxKey].shift();
+		this.removeEmptyPriority(maxKey);
+
+		return poppedValue
 	};
 
+	/**
+	 * Returns the number of elements in queue
+	 * @returns {Number} 
+	 */
 	this.length = function() {
 		return this.count;
-	}
-}
+	};
+};
 
-PriorityQueue.prototype.get_all_priorities = function() {
-	return Object.keys(this.store);
-}
+/**
+ * Returns all the priorities in descending order
+ * @returns {Array}
+ */
+PriorityQueue.prototype.getAllPriorities = function() {
+	return Object.keys(this.queue).sort(function(a, b){
+		return b - a;
+	});
+};
 
-// iterates through all the queue elements in priority-then-FIFO order
-PriorityQueue.prototype.forEach = function(callback) {
-	var keys = Object.keys(this.store).sort();
+/**
+ * Iterates through all the queue elements in priority-then-FIFO order
+ * 
+ * @param {function} callback - Callback function to be used on every element
+ */
+PriorityQueue.prototype.priorityForEach = function(callback) {
+	var allPriorities = this.getAllPriorities();
+	allPriorities.forEach(function(priority) {
+		for (var valueIndex = 0; valueIndex < this.queue[priority].length; valueIndex++) {
+			callback(this.queue[priority][valueIndex]);
+		}
+	}, this);

```

Comment:
Passing the context is required, else the `this` inside of the `forEach()` would be the global `window`